src/main/kotlin/com/demonwav/mcdev/i18n/actions/ConvertToTranslationAction.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.actions
+
+import com.demonwav.mcdev.i18n.intentions.ConvertToTranslationIntention
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.psi.PsiLiteral
+
+class ConvertToTranslationAction : AnAction() {
+    override fun actionPerformed(e: AnActionEvent) {
+        val file = e.getData(LangDataKeys.PSI_FILE) ?: return
+        val editor = e.getData(PlatformDataKeys.EDITOR) ?: return
+        val element = file.findElementAt(editor.caretModel.offset) ?: return
+        ConvertToTranslationIntention().invoke(editor.project ?: return, editor, element)
+    }
+
+    override fun update(e: AnActionEvent) {
+        val file = e.getData(LangDataKeys.PSI_FILE)
+        val editor = e.getData(PlatformDataKeys.EDITOR)
+        if (file == null || editor == null) {
+            e.presentation.isEnabled = false
+            return
+        }
+        val element = file.findElementAt(editor.caretModel.offset)
+        e.presentation.isEnabled = element != null && element.parent is PsiLiteral && (element.parent as PsiLiteral).value is String
```

Comment:
Not sure if this makes it harder to read but you could write this as
`(element?.parent as? PsiLiteral)?.value is String`